### PR TITLE
Bug 1852473: kubevirt: fix importing VMs with same vnicIDs for ovirt provider

### PR DESCRIPTION
**Pull request by suomiy**
- don't allow Pod networks for such nics
- allow only same network values for nics with same vnicID

### CODE REVIEW

**Comment by suomiy** [comment]
nics with same nic profiles:

net1

editing:

net2


after edit both get changed
net3

and vice versa (works in any direction)

net4

net5

please review @yaacov @pcbailey @irosenzw @glekner

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/providers/ovirt/ovirt-prefill-vm.ts [hunk 1]
```diff
@@ -90,15 +90,25 @@ export const getDisks = (vm: OvirtVM, storageClassConfigMap: ConfigMapKind): VMW
 
 export const getNics = (vm: OvirtVM): VMWizardNetwork[] => {
   const getUniqueName = createUniqueNameResolver(vm.nics);
-  const nics = (vm.nics || []).filter((n) => n);
+  const nics = (vm.nics || []).filter(
+    (nic) => nic && nic.name && nic.vnicid != null && nic.vnicid !== '',
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/providers/ovirt/ovirt-prefill-vm.ts:94]: can we use the ? notation here ?
`nic?. ...1`

File: frontend/packages/kubevirt-plugin/src/components/modals/nic-modal/nic-modal.tsx [hunk 2]
```diff
@@ -253,7 +263,11 @@ export const NICModal = withHandlePromise((props: NICModalProps) => {
             network={resultNetwork}
             onChange={onNetworkChoiceChange}
             nads={nads}
-            allowPodNetwork={allowPodNetwork}
+            allowPodNetwork={
+              editConfig?.allowPodNetworkOverride != null
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/modals/nic-modal/nic-modal.tsx:267]: does `editConfig?.allowPodNetworkOverride || allowPodNetwork` makes sense here ?

Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/modals/nic-modal/nic-modal.tsx:267]: what do you mean by "make sense"? 
Pod network is allowed only for one nic in kubevirt, so we have to disable it here for duplicate networks.

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/providers/ovirt/ovirt-prefill-vm.ts (hunk 1 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/providers/ovirt/ovirt-prefill-vm.ts]: fixed

File: frontend/packages/kubevirt-plugin/src/components/modals/nic-modal/nic-modal.tsx (hunk 2 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/modals/nic-modal/nic-modal.tsx:267]: > what do you mean by "make sense"?
`a != null ? a : b`
can be `a || b`

**Pull request merged**